### PR TITLE
Vulcan: Problem List: Ancestor problem lists

### PR DESCRIPTION
**Pull request by ianrohde**
## Issue

Add ancestor Problems to Problems List page when available. See [issue](https://github.com/iFixit/ifixit/issues/51062) description for further details.

> Mock-up text is not final. Figure out final copy with Max.

Text will be updated separately in https://github.com/iFixit/ifixit/issues/51815

## CR/QA

`/Troubleshooting/dryer` (no ancestors)
`/Troubleshooting/AEG_T865901H_dryer` (has ancestors)

Closes https://github.com/iFixit/ifixit/issues/51062

### CODE REVIEW

**Review by hcutrone — APPROVED** [review]
CR 🌯 Looks good to me!

**Comment by kthaler** [comment]
QA 💎 - When a device has problem pages, and it's ancestor also has problem pages, the device's troubleshooting page will display it's own problem pages, and in a second heading it will show it's ancestor's problem pages. 

deploy_block 🦖 

1. What is expected when there are a hierarchy of children with problem pages? Should a child-of-a-child device inherit the base problem pages or it's own parent problem pages, or both? For example, I made a child of the [AEG T865901H dryer](https://kthaler.cominor.com/Device/AEG_T865901H_dryer) device and gave it a troubleshooting page: https://me.cominor.com/Device/child_test . In the troubleshooting section of the device page, only the dryer troubleshooting pages are inherited, and not the AEG T865901H dryer troubleshooting page. In the [Troubleshooting page of the child device](https://react-commerce-git-problem-list-ancestor-problem-lists-ifixit.vercel.app/Troubleshooting/child_test), the AEG T865901H dryer Troubleshooting page isn't there, only the dryer pages. 


2.

>Text will be updated separately in https://github.com/iFixit/ifixit/issues/51815

Not sure if my issue applies then, so please disregard if this isn't in the scope of the pull. 
At widths of 767px to 575px, the h2 heading seems too close to the navbar. Video:

https://github.com/iFixit/react-commerce/assets/145375039/2afcfb7b-d2ad-40c2-a61f-4b6ca6bead43

**Comment by ianrohde** [comment]
> At widths of 767px to 575px, the h2 heading seems too close to the navbar.

This should be corrected, thanks!

**Comment by andyg0808** [comment]
> What is expected when there are a hierarchy of children with problem pages? Should a child-of-a-child device inherit the base problem pages or it's own parent problem pages, or both?

The troubleshooting pages of a device with the inheritance flag are inherited by all its child devices. No pages from child devices are inherited by more distant descendants.
For full details, here's the spec: https://github.com/iFixit/ifixit/issues/48260

>For example, I made a child of the [AEG T865901H dryer](https://kthaler.cominor.com/Device/AEG_T865901H_dryer) device and gave it a troubleshooting page: [me.cominor.com/Device/child_test](https://me.cominor.com/Device/child_test). In the troubleshooting section of the device page, only the dryer troubleshooting pages are inherited, and not the AEG T865901H dryer troubleshooting page.

:+1: That's what we want. We can add the inheritance flag to the AEG T865901H dryer to have its troubleshooting pages also get inherited, at which point the `child_test` device should have both Dryer and AEG T865901H troubleshooting pages.

un_deploy_block :baby:

**Review by hcutrone — DISMISSED** [review]
CR 🌯

**Comment by deltuh-vee** [comment]
QA 🎬 
h2 heading is no longer too close to the navbar.
Did you also mean to make this larger?
CleanShot 2024-02-07 at 10 22 03@2x

**Comment by ianrohde** [comment]
@deltuh-vee was that observed on the latest preview? I'm not seeing it: https://react-commerce-prod-gi2hw5yqz-ifixit.vercel.app/Troubleshooting/dryer

**Comment by ianrohde** [comment]
Answers heading fontSize fixed in https://github.com/iFixit/react-commerce/pull/2231/commits/35dec8c6c55dadbc01ebe963cfa2670391899fe8

un_deploy_block 🟢

**Comment by deltuh-vee** [comment]
QA 🎬 
Larger text has been fixed.

File: frontend/app/(defaultLayout)/Troubleshooting/[device]/components/Answers.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ export function Answers({
                <Heading
                   as={hasProblems ? 'h4' : 'h3'}
                   color="gray.800"
-                  fontSize={{ base: '24px', sm: '30px' }}
+                  fontSize={{ base: '20px', mdPlus: '24px' }}
```
Comment (hcutrone) [inline frontend/app/(defaultLayout)/Troubleshooting/[device]/components/Answers.tsx:20]: Is `mdPlus` a custom breakpoint we created? I haven't seen it before

Comment (ianrohde) [reply frontend/app/(defaultLayout)/Troubleshooting/[device]/components/Answers.tsx:20]: Yeah, I added it here: https://github.com/iFixit/react-commerce/pull/2143/files#diff-9e3cebe6686b9c7a2c8f51158b777b7f8d7b7bd53fd8a126776adc1e655bf956R7

tl;dr: Chakra uses min-width @media queries, so we need 'md' + 1px
